### PR TITLE
Checkout

### PR DESCRIPTION
**Pull request by johnson-alan**
### Assignee Tasks

- [ ] added unit tests (or none needed)
- [ ] written relevant docs (or none needed)
- [ ] referenced any relevant issues (or none exist)

### Guidelines

Please add a description of this Pull Request's motivation, scope, outstanding issues or potential alternatives, reasoning behind the current solution, and any other relevant information for posterity.

---

*Your PR Notes Here*
#10

### CODE REVIEW

File: client/components/Checkout.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+const mapState = ({ cart, user, checkout }) => ({ cart, user, checkout })
+const mapDispatch = (dispatch) => {
+  return {
+    getCart(userId) { dispatch(fetchCart(userId)) },
+    handleShippingFormSubmit(values) {
+      values.completedShippingForm = true
+      dispatch(addShippingDetails(values))
+    },
+    handleBillingFormSubmit() {
+      // dispatch(updateOrders(orderId, status, orders))
```
Comment (schai6) [inline client/components/Checkout.js]: remove this

File: client/routes.js [hunk 2]
```diff
@@ -3,9 +3,15 @@ import { connect } from 'react-redux'
 import { Route, Switch, Router } from 'react-router-dom'
 import PropTypes from 'prop-types'
 import history from './history'
-import { Main, Login, Signup, UserHome, VisitorHome, Cart, AllWatches, SingleWatch, AllOrders, SingleOrder } from './components'
+import { Main, Login, Signup, UserHome, VisitorHome, Cart, Checkout, AllWatches, SingleWatch, AllOrders, SingleOrder } from './components'
+<<<<<<< HEAD
```
Comment (schai6) [inline client/routes.js]: merge conflict

File: client/routes.js [hunk 3]
```diff
@@ -33,13 +39,13 @@ class Routes extends Component {
...
                   {/* Routes placed here are only available after logging in */}
                   <Route path="/home" component={UserHome} />
-                  <Route path="/cart" component={Cart} />
                   <Route exact path="/orders" component={AllOrders} />
                   <Route path="/orders/:orderId" component={SingleOrder} />
                 </Switch>
               }
               {/* Display landing page as fallback */}
               <Route path="/cart" component={Cart} />
+              <Route patn="/checkout" component={Checkout} />
```
Comment (schai6) [inline client/routes.js]: patn should be path

File: client/store/cart.js [hunk 4]
```diff
@@ -23,6 +24,21 @@ export const showQuantityUnderflowError = watchId => ({type: SHOW_QUANTITY_UNDER
...
+    if (userId) {
+      axios.put(`/api/users/${userId}/new`)
+      .then(res => res.data)
+      .then(() => {
+        dispatch(emptyCart())
+      })
+    } else {
+      dispatch(emptyCart())
+    }
+    console.log('history', history)
```
Comment (schai6) [inline client/store/cart.js]: remove log

File: public/style.css [hunk 5]
```diff
@@ -108,3 +106,28 @@ form div {
...
+  display: flex;
+  flex-wrap: wrap;
+  padding: 3em 4em 3em 3em;
+  border: 1px solid rgba(34,36,38,.15);
+  border-radius: .28571429rem;
+}
+
+.checkout-form div {
+  flex-basis: 100%;
+<<<<<<< HEAD
```
Comment (schai6) [inline public/style.css]: merge conflict

**Review by schai6 — CHANGES_REQUESTED** [review]
Just a few small things!

**Comment by jluria** [comment]
@allenhj Not meaning to step on your toes. Just glanced before going to bed, and saw that a few things could be quickly addressed, and maybe help to merge and deploy in the morning. Left the couple things that maybe had been intentional on your part or you wanted to address

**Comment by jluria** [comment]
@allenhj Congrats on PR number 100!!

**Comment by johnson-alan** [comment]
I think I addressed everything. Thanks Josh and Sam

**Comment by johnson-alan** [comment]
These changes break viewing the cart for a logged in user for some reason. Trying to figure out why

**Comment by johnson-alan** [comment]
It was an issue with routing order. Fixed it and pushing changes to the remote now

**Review by schai6 — APPROVED** [review]

**Pull request merged**